packages/rocketchat-2fa/client/TOTPSaml.js

```diff
@@ -0,0 +1,33 @@
+Meteor.loginWithSamlAndTOTP = function(options, code, callback) {
+	// support a callback without options
+	if (!callback && typeof options === 'function') {
+		callback = options;
+		options = null;
+	}
+
+	if (!options) {
+		options = {};
+		options.credentialToken = `id-${ Random.id() }`;
+	}
+	const { credentialToken } = options;
+
+	Accounts.saml.initiateLogin(options, function(/* error, result */) {
+		Accounts.callLoginMethod({
+			methodArguments: [{
+				saml: true,
```

Comment:
Isn't this inverted? `totp` property shouldn't be the top one?

Reply:
All three properties are on the same object on the first index of the array, the property order doesn't matter.

Reply:
I wasn't talking about the order, but something like:
```
methodArguments: [{
	totp: {
		code,
		login: {
			saml: true,
			credentialToken,
		}
	},
}],
```